### PR TITLE
Added final newline formatter

### PR DESCRIPTION
**Pull request by JoeRobich**
Adds a new formatter for the `insert_final_newline` editorconfig option.

Closes https://github.com/dotnet/format/issues/111

### CODE REVIEW

File: src/Formatters/EOFNewlineFormatter.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.CodingConventions;
+
+namespace Microsoft.CodeAnalysis.Tools.Formatters
+{
+    internal sealed class EOFNewLineFormatter : DocumentFormatter
```
Comment (sharwell) [inline src/Formatters/EOFNewlineFormatter.cs]:
```suggestion
    internal sealed class EndOfFileNewLineFormatter : DocumentFormatter
```

**Comment by JoeRobich** [comment]
@sharwell I had assumed that we were letting the Editor handle some of these core .editorconfig options.  I can add it there instead.

**Comment by sharwell** [comment]
Not all options would apply there, but it seems these two would.

**Comment by JoeRobich** [comment]
@sharwell I see that Roslyn already has an issue to support final newline in the formatter (https://github.com/dotnet/roslyn/issues/33468). Going to merge this and we can remove when the functionality is implemented in Roslyn.

**Pull request merged**